### PR TITLE
Fix: note creation / deletion should respect doc permissions

### LINKED ISSUE #3902: [BUG] User without write permissions can add note
### Description

The user is able to add notes to documents who that user only has view permissions to.

### Steps to reproduce

1. Set owner to User_A on a document.
2. Set User_B, or a group User_B is member of to have view-permission on the document
3. User_B can view as normal, can not edit Details or Content, but can add Notes.

The note-add wil load forever(?) until user hit F5 and the note is there.
image
*F5*
image

RESULT:
The note is stored for the document, but just the user itself can see ut, even a superuser is not able to view that note.

COMMENT: I don't know if this is by design, but with that error in the log i guess not.

### Webserver logs

```bash
[2023-07-31 21:24:27,569] [WARNING] [paperless.api] An error occurred saving note: You do not have permission to perform this action.
```

### Browser logs

_No response_

### Paperless-ngx version

1.16.5

### Host OS

UnRAID,

### Installation method

Docker - official image

### Browser

Google Chrome

### Configuration changes

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by shamoon**
## Proposed change

See linked issue. The button should have been disabled but I also made the API more explicitly respect the doc perms, see tests. Of note, the note in the linked issue wasn't actually getting created, that was just a frontend cache kind thing, but still could be handled better.

Fixes #3902

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain):

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [ ] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by shamoon** [comment]
I have to look at why it says cov decreased, I thought I covered it

File: src/documents/tests/test_api.py [hunk 1]
```diff
@@ -2369,6 +2369,55 @@ def test_create_note(self):
...
+        )
+
+        response = self.client.delete(
+            f"/api/documents/{doc.pk}/notes/?id={note.pk}",
+            format="json",
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
     def test_delete_note(self):
```
Comment (stumpylog) [inline src/documents/tests/test_api.py:2421]: Do you want to add testing for the delete that is permissions aware?

Comment (shamoon) [reply src/documents/tests/test_api.py:2421]: TBH, I thought that this did (user1 tries to delete a note from user2's doc)
https://github.com/paperless-ngx/paperless-ngx/blob/49b915eb3300e11a087e3a7cf6283a29ec1c7996/src/documents/tests/test_api.py#L2408-L2419

My pytest setup seems randomly broken at the moment so Im a little hobbled

Comment (shamoon) [reply src/documents/tests/test_api.py:2421]: TBH, I thought that this did (user1 tries to delete a note from user2's doc)
https://github.com/paperless-ngx/paperless-ngx/blob/49b915eb3300e11a087e3a7cf6283a29ec1c7996/src/documents/tests/test_api.py#L2408-L2419

My pytest setup seems randomly broken at the moment so Im a little hobbled

Comment (stumpylog) [reply src/documents/tests/test_api.py:2421]: It looks like `view_document` doesn't allow a user to view the notes on a document.  I'm not sure if it should.  But I think the delete gets blocked earlier up the chain and doesn't reach this.

Comment (shamoon) [reply src/documents/tests/test_api.py:2421]: Cheers stumpy, got me on the right track! Should be sorted (as for pytest, I'll live to fight that one another day 🙃)

**Review by stumpylog — APPROVED** [review]

**Pull request merged**